.github/workflows/ci_upgrade.yml

```diff
@@ -0,0 +1,162 @@
+name: JDBC Upgrade Tests
+on:
+  push:
+    branches:
+      - BABEL_1_X_DEV
+  pull_request:
+    branches:
+      - BABEL_1_X_DEV
+
+jobs:
+  extension-tests:
+    env:
+      ENGINE_VER_FROM: BABEL_1_0_0__PG_13_4
+      EXTENSION_VER_FROM: BABEL_1_0_0
+      ENGINE_VER_TO: BABEL_1_X_DEV__PG_13_6
+      EXTENSION_VER_TO: BABEL_1_X_DEV
+
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Clone, build, and run tests for Postgres engine using ${{env.ENGINE_VER_FROM}}
+        run: |
+          cd ..
+          git clone --branch $ENGINE_VER_FROM https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish.git
+          sudo apt-get update
+          sudo apt-get install uuid-dev openjdk-8-jre libicu-dev libxml2-dev openssl libssl-dev python-dev libossp-uuid-dev libpq-dev cmake pkg-config g++ build-essential bison 
+          cd postgresql_modified_for_babelfish
+          ./configure --prefix=$HOME/postgres/ --with-python PYTHON=/usr/bin/python2.7 --enable-debug CFLAGS="-ggdb" --with-libxml --with-uuid=ossp --with-icu
```

Comment:
We can modify build-modified-postgres composite action to take branch as an argument so that we can re-use that action to build engine for any version.

Reply:
Done. `build-modified-postgres` composite action can now take an input `engine-version` optionally.